ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/History/EraParams.hs

```diff
@@ -136,17 +146,65 @@ data EraParams = EraParams
   , eraSlotLength :: !SlotLength
   , eraSafeZone :: !SafeZone
   , eraGenesisWin :: !GenesisWindow
+  , eraPerasRoundLength :: !(PerasEnabled PerasRoundLength)
```

Comment:
Why not using `Maybe` instead? This add cognitive load, as the reader now has to know what `PerasEnabled` is.

Reply:
I don't have a strong opinion about this but, if we used a `Maybe`, then a programmer would still have to reach out to the docs/source code to see what that optional value is encoding. This might be just a tad more self-documenting than that.

Reply:
Initially I used `StrictMaybe`, but in one of the review rounds Alex suggested we used a custom type as it was more descriptive at call-sites.

In tests, I also had to follow a pattern of wrapping the era end into another `Maybe` (see [here](https://github.com/IntersectMBO/ouroboros-consensus/blob/1be66fb83180909e57d678f05460abd143f9b1aa/ouroboros-consensus/test/consensus-test/Test/Consensus/HardFork/Summary.hs#L197), which would be two layers of `Maybe`.

I think we have several such small custom types in the code base, so I though it would be alright. I can refactor it back to be a `Maybe` if you have a strong opinion.